pkg/agent/common/backoff/size_backoff.go

```diff
@@ -0,0 +1,59 @@
+package backoff
+
+// SizeLimitedBackOff defines interface for implementing a size based backoff for requests which
+// contain number of records to be processed by server.
+type SizeLimitedBackOff interface {
+	// NextBackOff returns the duration to wait before retrying the operation,
+	// or backoff.
+	NextBackOff() int
+
+	// Success indicates the backoff implementation that previous request succeeded
+	// so that it can adjust backoff accordingly for next request.
+	Success()
+
+	// Failure indicates the backoff implementation that previous request failed
+	// so that it can adjust backoff accordingly for next request.
+	Failure()
+
+	// Reset to initial state.
+	Reset()
+}
+
+type sizeLimitedBackOff struct {
+	currentSize int
+	maxSize     int
+}
+
+var _ SizeLimitedBackOff = (*sizeLimitedBackOff)(nil)
+
+func (r *sizeLimitedBackOff) NextBackOff() int {
+	return r.currentSize
+}
+
+func (r *sizeLimitedBackOff) Success() {
+	r.currentSize *= 2
+	if r.currentSize > r.maxSize {
+		r.currentSize = r.maxSize
+	}
+}
+
+func (r *sizeLimitedBackOff) Failure() {
+	r.currentSize /= 2
+	if r.currentSize < 1 {
+		r.currentSize = 1
+	}
+}
```

Comment:
we're applying calculations to `r.currentSize` and then making a correction based on max / min values. we *probably* don't need to worry about race conditions, but having a single assignment to the struct in the function would still at least put me more at ease

```suggestion
	newSize := r.currentSize / 2
	if newSize < 1 {
		newSize = 1
	}
	r.currentSize = newSize
}
```

Reply:
done